### PR TITLE
Try out cibuildwheel.

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

This is enabled for push & pull_request right now, but since we have nowhere to upload anything permanent, I'll probably switch this to tags-only for final usage, if merged.

## PR Checklist

- [n/a] Has Pytest style unit tests
- [n/a] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [n/a] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [n/a] Documented in doc/api/next_api_changes/* if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by QuLogic** [comment]
Since enabling this on `push` means it builds on my fork also, I'm trying this out on [a separate branch](https://github.com/QuLogic/matplotlib/tree/cibuildwheel2) so as to not clog up our other CI.

**Comment by QuLogic** [comment]
This should now be building on all 3 platforms, but it doesn't run any tests. It's also doing manylinux2010 instead of manylinux1, though I'm sure that could be tweaked. GitHub Actions doesn't do arm or s390x, I don't think, so we'd still need the matplotlib-wheels repo to build those.

File: .github/workflows/cibuildwheel.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Build CI wheels
+
+on: [push, pull_request]
```
Comment (tacaswell) [inline .github/workflows/cibuildwheel.yml:3]: change to tag?

Comment (tacaswell) [reply .github/workflows/cibuildwheel.yml:3]: or, tag + push

**Pull request merged**